dali/test/python/test_dali_variable_batch_size.py

```diff
@@ -0,0 +1,622 @@
+
+
+ops_image_default_args = [
+    fn.brightness,
+    fn.brightness_contrast,
+    fn.cat,
+    fn.color_twist,
+    fn.contrast,
+    fn.copy,
+    fn.crop_mirror_normalize,
+    fn.dump_image,
+    fn.hsv,
+    fn.hue,
+    fn.old_color_twist,
+    fn.reductions.mean,
+    fn.reductions.mean_square,
+    fn.reductions.rms,
+    fn.reductions.min,
+    fn.reductions.max,
+    fn.reductions.sum,
+    fn.saturation,
+    fn.shapes,
+    fn.sphere,
+    fn.stack,
+    fn.water,
+]
+
+
+def test_ops_image_default_args():
+    for op in ops_image_default_args:
+        yield image_data_helper, op
+
+
+ops_image_custom_args = [
+    {'operator_fn': fn.cast, 'dtype': types.INT32},
+    {'operator_fn': fn.color_space_conversion, 'image_type': types.BGR, 'output_type': types.RGB},
+    {'operator_fn': fn.coord_transform, 'M': .5, 'T': 2},
+    {'operator_fn': fn.crop, 'crop': (5, 5)},
+    {'operator_fn': fn.erase, 'anchor': [0.3], 'axis_names': "H", 'normalized_anchor': True,
+     'shape': [0.1], 'normalized_shape': True},
+    {'operator_fn': fn.fast_resize_crop_mirror, 'crop': [5, 5], 'resize_shorter': 10,
+     'devices': ['cpu']},
+    {'operator_fn': fn.flip, 'horizontal': True},
+    {'operator_fn': fn.gaussian_blur, 'window_size': 5},
+    {'operator_fn': fn.normalize, 'batch': True},
+    {'operator_fn': fn.pad, 'fill_value': -1, 'axes': (0,), 'shape': (10,)},
+    {'operator_fn': fn.paste, 'fill_value': 69, 'ratio': 1, 'devices': ['gpu']},
+    {'operator_fn': fn.reinterpret, 'rel_shape': [0.5, 1, -1]},
+    {'operator_fn': fn.resize, 'resize_x': 50, 'resize_y': 50},
+    {'operator_fn': fn.resize_crop_mirror, 'crop': [5, 5], 'resize_shorter': 10,
+     'devices': ['cpu']},
+    {'operator_fn': fn.rotate, 'angle': 25},
+    {'operator_fn': fn.transpose, 'perm': [2, 0, 1]},
+    {'operator_fn': fn.warp_affine, 'matrix': (.1, .9, 10, .8, -.2, -20)},
+]
+
+
+def test_ops_image_custom_args():
+    for op in ops_image_custom_args:
+        print("\nTesting ", op)
+        image_data_helper(**op)
+
+
+float_array_ops = [
+    {'operator_fn': fn.power_spectrum, 'devices': ['cpu']},
+    {'operator_fn': fn.preemphasis_filter},
+    {'operator_fn': fn.spectrogram, 'nfft': 60, 'window_length': 50, 'window_step': 25},
+    {'operator_fn': fn.to_decibels},
+]
+
+
+def test_float_array_ops():
+    for op in float_array_ops:
+        print("\nTesting ", op)
+        float_array_helper(**op)
+
+
+random_ops = [
+    {'operator_fn': fn.jitter, 'devices': ['gpu']},
+    {'operator_fn': fn.random_resized_crop, 'size': 69},
+]
+
+
+def test_random_ops():
+    for op in random_ops:
+        print("\nTesting ", op)
+        random_op_helper(**op)
+
+
+def test_batch_permute():
+    def pipe(max_batch_size, input_data, device):
+        pipe = Pipeline(batch_size=max_batch_size, num_threads=4, device_id=0)
+        perm = fn.batch_permutation(seed=420)
+        data = fn.external_source(source=input_data, num_outputs=1, cycle=False, device=device)
```

Comment:
```suggestion
        data = fn.external_source(source=input_data, cycle=False, device=device)
```
Then  your output will not be a tuple. You can remove all the `num_outputs=1` occurrences.

Reply:
Done